src/RestServer/Extensions/LedgerContractExtensions.cs

```diff
@@ -0,0 +1,144 @@
+// Copyright (C) 2015-2023 The Neo Project.
+//
+// The Neo.Plugins.RestServer is free software distributed under the MIT software license,
+// see the accompanying file LICENSE in the main directory of the
+// project or http://www.opensource.org/licenses/mit-license.php
+// for more details.
+//
+// Redistribution and use in source and binary forms with or without
+// modifications are permitted.
+
+using Neo.IO;
+using Neo.Persistence;
+using Neo.Plugins.RestServer.Models.Blockchain;
+using Neo.SmartContract;
+using Neo.SmartContract.Native;
+using Neo.Wallets;
+
+#pragma warning disable IDE0060
+
+namespace Neo.Plugins.RestServer.Extensions
+{
+    internal static class LedgerContractExtensions
+    {
+        private const byte _prefix_block = 5;
+        private const byte _prefix_transaction = 11;
+        private const byte _prefix_account = 20;
+        private const byte _prefix_totalsupply = 11;
+
+        public static IEnumerable<(StorageKey key, StorageItem value)> GetStorageByPrefix(this ContractState contractState, DataCache snapshot, byte[] prefix)
+        {
+            ArgumentNullException.ThrowIfNull(nameof(contractState));
+            ArgumentNullException.ThrowIfNull(nameof(snapshot));
+            if (prefix?.Length == 0)
+                throw new ArgumentNullException(nameof(prefix));
+            foreach (var (key, value) in snapshot.Find(StorageKey.CreateSearchPrefix(contractState.Id, prefix)))
+                yield return (key, value);
+        }
+
+        public static StorageItem GetStorageByKey(this ContractState contractState, DataCache snapshot, byte[] storageKey)
+        {
+            ArgumentNullException.ThrowIfNull(nameof(contractState));
+            ArgumentNullException.ThrowIfNull(nameof(snapshot));
+            if (storageKey?.Length == 0)
+                throw new ArgumentNullException(nameof(storageKey));
+            foreach (var (key, value) in snapshot.Find(StorageKey.CreateSearchPrefix(contractState.Id, storageKey)))
+                if (key.Key.Span.SequenceEqual(storageKey))
+                    return value;
+            return default;
+        }
```

Comment:
Some of these methos are not used, it can be removed

Reply:
I'm not sure. I think `ContractController` uses for contract storage requests.